plugins/plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/navigation/filestructure/FileStructurePresenter2.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2012-2017 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.ide.ext.java.client.navigation.filestructure;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import org.eclipse.che.ide.api.editor.EditorPartPresenter;
+import org.eclipse.che.ide.api.editor.texteditor.TextEditor;
+import org.eclipse.che.ide.api.resources.VirtualFile;
+import org.eclipse.che.ide.ext.java.client.service.JavaLanguageExtensionServiceClient;
+import org.eclipse.che.ide.ext.java.dto.DtoClientImpls;
+import org.eclipse.che.ide.ui.loaders.request.LoaderFactory;
+import org.eclipse.che.ide.ui.loaders.request.MessageLoader;
+import org.eclipse.che.ide.util.loging.Log;
+import org.eclipse.che.jdt.ls.extension.api.dto.ExtendedSymbolInformation;
+import org.eclipse.che.jdt.ls.extension.api.dto.FileStructureCommandParameters;
+import org.eclipse.che.plugin.languageserver.ide.filestructure.ElementSelectionDelegate;
+import org.eclipse.che.plugin.languageserver.ide.util.OpenFileInEditorHelper;
+
+/**
+ * Manages jdt.ls based file structure view
+ *
+ * @author Thomas Mäder
+ */
+@Singleton
+public class FileStructurePresenter2
```

Comment:
Can you give a better name?

Reply:
Also it would be good to have junit test for it as it is done for current `FileStructurePresenter`

Reply:
@evidolob the idea is to rename it to FileStructurePresenter once we're done with the migration
@bdshadow noted.

Reply:
@evidolob the idea is to rename it once we delete the "old" version.